app/Enum/StatusVisibility.php

```diff
@@ -3,11 +3,43 @@
 
 namespace App\Enum;
 
-enum StatusVisibility: int
+use App\Interfaces\IconEnumInterface;
+
+enum StatusVisibility: int implements IconEnumInterface
 {
     case PUBLIC = 0;
     case UNLISTED = 1;
     case FOLLOWERS = 2;
     case PRIVATE = 3;
     case AUTHENTICATED = 4;
+
+    public function faIcon(): string {
+        return match ($this) {
+            self::PUBLIC        => 'fa-globe-americas',
+            self::UNLISTED      => 'fa-lock-open',
+            self::FOLLOWERS     => 'fa-user-friends',
+            self::PRIVATE       => 'fa-lock',
+            self::AUTHENTICATED => 'fa-user-check'
+        };
+    }
+
+    public function title(): string {
+        return match ($this) {
+            self::PUBLIC        => __('status.visibility.0'),
+            self::UNLISTED      => __('status.visibility.1'),
+            self::FOLLOWERS     => __('status.visibility.2'),
+            self::PRIVATE       => __('status.visibility.3'),
+            self::AUTHENTICATED => __('status.visibility.4')
+        };
```

Comment:
```suggestion
        return __('status.visibility.' . $this->value);
```

Wouldn't this be easier? (untested)

Reply:
Yes but I'd like to increase readability and maybe change the name of the translation in the future (visibility.0 is very ambiguous)